internal/util/search/search.go

```diff
@@ -211,3 +217,57 @@ func (sr *SearchReplace) shouldPutLiteralByPath() bool {
 		sr.ByValueRegex == "" &&
 		sr.PutLiteral != ""
 }
+
+// setterPatternFromRegex takes the field value of a node, valueRegex provided by
```

Comment:
So what happens if we can't derive the setter values from the pattern? Does the user get an error or do we just use the pattern directly?
We should clearly define how we do this and which situations we will not be able to determine the setter values.

Reply:
Yes. Described it in lines 225-228 below with example.